src/index.js

```diff
@@ -1 +1,32 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
+  }
+
+  prefix(request) {
+    return request;
+  }
+
+  postfix(response) {
+    return response;
+  }
+
+  send(url, options) {
```

Comment:
How are you expecting this to be used? I think that while the prefix/postfix is a nice feature, this doesn't go far enough in DRYing a client's code.

Remember, the requests in a client have an expected signature of `request(options, query)`. `options` is this case are things needed by the client, such as an id for a specific resource, POST data, etc. It's unrelated to the `options` in fetch's Request.

With what you have now, the clients will have to parse the querystring onto the `url` rather than letting stipulate handle it, as well as attaching the `body` to the `options` if it's present.  I don't think the latter is too terrible, but it does need to do null/undefined checks, but url parsing will need to be DRY too.

I think it's worth considering a signature of `send(url, options, body, query)` that takes client knowledge and converts it into fetch knowledge; else, the clients are going to have to do the conversion too.

Take a look at yosbbejs and consider how the requests would be rewritten to use stipulate.